### ISSUE TITLE
Add timeout support

### ISSUE THREAD

**Issue by introom** [justbur/emacs-which-key#295]
Is it possible to add timeout support so that when which-key display is idle for 3 second, it is automatically cancelled?

**Comment by justbur** [comment]
Yes, it's possible, but why would you want to remain idle in the middle of a key sequence? Or is this is another situation?

**Comment by introom** [comment]
Some special case though, such as tmux sending keys  (C-l) to all panes to clear the pane. And it happened that emacs is running in some pane. We want to cancel the C-l.

**Comment by justbur** [comment]
But a which-key timeout won't cancel the key sequence. which-key is passive. It doesn't manipulate the key sequences
